java/common/exception/ErrorMessage.java

```diff
@@ -79,14 +79,12 @@ public class ErrorMessage extends com.vaticle.typedb.common.exception.ErrorMessa
             new ErrorMessage(28, "Rule '%s' 'when' has not been provided with any patterns.");
     public static final ErrorMessage INVALID_RULE_WHEN_NESTED_NEGATION =
             new ErrorMessage(29, "Rule '%s' 'when' contains a nested negation.");
-    public static final ErrorMessage INVALID_RULE_WHEN_CONTAINS_DISJUNCTION =
-            new ErrorMessage(30, "Rule '%s' 'when' contains a disjunction.");
     public static final ErrorMessage INVALID_RULE_THEN =
             new ErrorMessage(31, "Rule '%s' 'then' '%s': must be exactly one attribute ownership, or exactly one relation.");
     public static final ErrorMessage INVALID_RULE_THEN_HAS =
             new ErrorMessage(32, "Rule '%s' 'then' '%s' tries to assign type '%s' to variable '%s', but this variable already had a type assigned by the rule 'when'. Try omitting this type assignment.");
     public static final ErrorMessage INVALID_RULE_THEN_VARIABLES =
-            new ErrorMessage(33, "Rule '%s' 'then' variables must be present in rule 'when'.");
+            new ErrorMessage(33, "Rule '%s' 'then' variables must be present in every branch of rule 'when'.");
```

Comment:
we don't have `branch` on TypeQL. You need to use terminology local to the domain. 
The validation should actually require that every `then` variable is bound outside outside of the disjunction before normalisation:
```
$x $y $z;
{ $x has $a; } or { $y has $a }
```
we only allow x, y, z to be used in the then

Reply:
` 'then' variables must be present in the 'when' outside of disjunctions & negations.`
OR
` 'then' variables must be present in the trunk of the 'when' `